README.md

```diff
@@ -378,6 +379,28 @@ pagination: {
 }
 ```
 
+### Sudo
+For private gitlab instances, administrators are able to impersonate users through the API. To do so, you have to set the 'Sudo' header on the services you want to impersonate the user for.
+
+For example, if you want to disable notifications for a specific user:
+```javascript
+import Gitlab from 'gitlab';
+
+const api = new Gitlab({
```

Comment:
Id suggest changing this to:

```javascript

const { NotificationSettings } = new Gitlab({
  url:   'http://example.com', // Defaults to http://gitlab.com
  token: 'abcdefghij123456' // Can be created in your profile
  sudo: userid_or_username
}) ;

....
```

Or if youre looking to add sudo for a specific api request, i should expose it differently then having to add/delete the prop from the headers.

Reply:
If so, ill make a MR tonight to fix this for the next version so it will be much clearer. I really want to avoid changing the internals of the services once they've been initialized.

Reply:
Nevermind, I just saw the merged [PR](https://github.com/jdalrymple/node-gitlab/pull/210/commits/4bf574c0c019fe42f6d77aab15d558b4cfbb321a) about the sudo header. I only checked my local version of the code (4.1.1) and not the master branch.

Reply:
The current release assumes that you init it with the service, but if youd rather have it on a singular call vs on every call of that service, i can make a PR to handle that in a cleaner fashion :)

Reply:
Well, I'm not against a cleaner way to do this !

Reply:
Created an example here #223